pkg/cidata/cidata.go

```diff
@@ -370,6 +372,54 @@ func GenerateISO9660(instDir, name string, y *limayaml.LimaYAML, udpDNSLocalPort
 	return iso9660util.Write(filepath.Join(instDir, filenames.CIDataISO), "cidata", layout)
 }
 
+// getHostTimezone tries to get host timezone from env or specific files
+func getHostTimezone() string {
+	if runtime.GOOS == "windows" {
+		logrus.Debug("skip get host timezone on Windows")
+		return ""
+	}
+
+	// 1. try to get timezone from env.TZ
+	tzEnv := os.Getenv("TZ")
+	if tzEnv != "" &&
+		// skip if TZ is in "std offset dst offset, rule" format
+		!strings.Contains(tzEnv, " ") {
+		return tzEnv
+	}
```

Comment:
I think trying to parse `TZ` is futile; there are just too many different possible formats, and different C runtime libraries implement them differently.

- [glib specification](https://www.gnu.org/software/libc/manual/html_node/TZ-Variable.html)
- [POSIX specification](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03)

Neither of them show any spaces in the "std offset dst offset" format, and I believe they are not even allowed optionally.

Furthermore, `cloud-init` only allows settings corresponding to zoneinfo entries:

> timezone: (string) The timezone to use as represented in /usr/share/zoneinfo

If we want to support importing the `TZ` environment variable, then I think it should be optional, and added to `/etc/environment`. There is some precent for doing this (we have logic for proxy settings), but this should be a separate issue, if we decide to do this at all. For this PR I think you should concentrate on synchronizing the `/etc/localtime` and `/etc/timezone` settings, and nothing else.